HighlightRegister should be renamed to HighlightRegistry.

megangardner opened the pull request:
#### 5c10221be14f825562feca679a0c80ccb1e513c2
<pre>
HighlightRegister should be renamed to HighlightRegistry.
<a href="https://bugs.webkit.org/show_bug.cgi?id=262522">https://bugs.webkit.org/show_bug.cgi?id=262522</a>
rdar://116262991

Reviewed by Aditya Keerthi and Richard Robinson.

HighlightRegister is the old version, need to rename to HighlightRegistry
to be compliant with he current spec.

* Source/WebCore/Modules/highlight/HighlightRegistry.cpp: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.cpp.
(WebCore::HighlightRegistry::initializeMapLike):
(WebCore::HighlightRegistry::setFromMapLike):
(WebCore::HighlightRegistry::clear):
(WebCore::HighlightRegistry::remove):
(WebCore::HighlightRegistry::setHighlightVisibility):
(WebCore::annotationHighlightKey):
(WebCore::HighlightRegistry::addAnnotationHighlightWithRange):
* Source/WebCore/Modules/highlight/HighlightRegistry.h: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.h.
(WebCore::HighlightRegistry::create):
(WebCore::HighlightRegistry::isEmpty const):
(WebCore::HighlightRegistry::highlightsVisibility const):
(WebCore::HighlightRegistry::map const):
(WebCore::HighlightRegistry::highlightNames const):
* Source/WebCore/Modules/highlight/HighlightRegistry.idl: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.idl.

Canonical link: <a href="https://commits.webkit.org/268873@main">https://commits.webkit.org/268873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3cbdeb77b9469e2695ca10e392cc895f0a3a0779

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20907 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21314 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21980 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22797 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19480 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21144 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24554 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21492 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20762 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21129 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20899 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18158 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23653 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18061 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18972 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25264 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19146 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19164 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23176 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19730 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16762 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18977 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18804 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5020 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23302 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19549 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Might be nice to link to the spec in the commit message: https://drafts.csswg.org/css-highlight-api/#registration

smfr reviewed (changes requested):
This is missing a bunch of places that use the old name (e.g. in Document.cpp)

megangardner committed:
HighlightRegister should be renamed to HighlightRegistry.
https://bugs.webkit.org/show_bug.cgi?id=262522
rdar://116262991

Reviewed by Aditya Keerthi and Richard Robinson.

HighlightRegister is the old version, need to rename to HighlightRegistry
to be compliant with he current spec.

* Source/WebCore/Modules/highlight/HighlightRegistry.cpp: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.cpp.
(WebCore::HighlightRegistry::initializeMapLike):
(WebCore::HighlightRegistry::setFromMapLike):
(WebCore::HighlightRegistry::clear):
(WebCore::HighlightRegistry::remove):
(WebCore::HighlightRegistry::setHighlightVisibility):
(WebCore::annotationHighlightKey):
(WebCore::HighlightRegistry::addAnnotationHighlightWithRange):
* Source/WebCore/Modules/highlight/HighlightRegistry.h: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.h.
(WebCore::HighlightRegistry::create):
(WebCore::HighlightRegistry::isEmpty const):
(WebCore::HighlightRegistry::highlightsVisibility const):
(WebCore::HighlightRegistry::map const):
(WebCore::HighlightRegistry::highlightNames const):
* Source/WebCore/Modules/highlight/HighlightRegistry.idl: Renamed from Source/WebCore/Modules/highlight/HighlightRegister.idl.

Canonical link: https://commits.webkit.org/268873@main
